distributed/worker.py

```diff
@@ -2451,7 +2428,9 @@ def meets_resource_constraints(self, key):
         return True
 
     async def _maybe_deserialize_task(self, ts):
+        assert ts.runspec is not None
         if not isinstance(ts.runspec, SerializedTask):
+            # FIXME: What is this case??
```

Comment:
Yeah, I don't know -- I updated it when adding `TaskState`.  I guessed that there was a possibility of the `runspec` already being deserialized?

Reply:
Currently the runspec is deserialized right before it is being transitioned to executing

https://github.com/dask/distributed/blob/3fb3d8c8c14800ff8f1cd76d0052985f355ecfcc/distributed/worker.py#L2491-L2496

and 

https://github.com/dask/distributed/blob/3fb3d8c8c14800ff8f1cd76d0052985f355ecfcc/distributed/worker.py#L2510-L2515

Therefore, I would assume that as this is deserialized for all tasks in the `executing` state. We currently only allow transitions from `executing` to `long-running` and `done`. Therefore, if we trust that all task transitions are properly performed using `Worker.transition`, this case should not be possible.

Unless, of course, the transitioning itself hits any kind of exception after deserializing, i.e. essentially in `Worker.execute` but as far as I understand it we are not robust to an exception there at the moment, anyhow.